pkgs/games/openra-mods/default.nix

```diff
@@ -0,0 +1,166 @@
+# Based on: pkgs/games/openra/default.nix
+# Based on: https://build.opensuse.org/package/show/home:fusion809/openra-ura
+{ stdenv, fetchFromGitHub, dos2unix, pkgconfig, makeWrapper
+, lua, mono, dotnetPackages, python
+, libGL, openal, SDL2
+, zenity ? null
+, mod
+, engine
+, engineMods ? []
+}:
+
+with stdenv.lib;
+
+let
+  path = makeBinPath ([ mono python ] ++ optional (zenity != null) zenity);
+  rpath = makeLibraryPath [ lua openal SDL2 ];
+
+in stdenv.mkDerivation rec {
+  name = "openra-${mod.name}-${mod.version}";
+
+  srcs = [
+    mod.src
+    engine.src
+  ];
+
+  sourceRoot = ".";
+
+  buildInputs = with dotnetPackages; [
+    FuzzyLogicLibrary
+    MaxMindDb
+    MaxMindGeoIP2
+    MonoNat
+    NewtonsoftJson
+    NUnit3
+    NUnitConsole
+    OpenNAT
+    RestSharp
+    SharpFont
+    SharpZipLib
+    SmartIrc4net
+    StyleCopMSBuild
+    StyleCopPlusMSBuild
+  ] ++ [
+    dos2unix
+    pkgconfig
+    makeWrapper
+    lua
+    mono
+    python
+    libGL
+    openal
+    SDL2
+  ];
+
+  postUnpack = ''
+    mv ${engine.src.name} mod
+    cd mod
+  '';
+
+  prePatch = ''
+    dos2unix Makefile
+  '';
+
+  patches = [ mod.makefilePatch ];
+
+  postPatch = ''
+    sed -i 's/^VERSION.*/VERSION = ${mod.version}/g' Makefile
+
+    dos2unix *.md
+
+    sed -i \
+      -e 's/^VERSION.*/VERSION = ${engine.version}/g' \
+      -e '/fetch-geoip-db/d' \
+      -e '/GeoLite2-Country.mmdb.gz/d' \
+      ${engine.src.name}/Makefile
+
+    sed -i 's|locations=.*|locations=${lua}/lib|' ${engine.src.name}/thirdparty/configure-native-deps.sh
+  '';
+
+  configurePhase = ''
+    make version VERSION=${escapeShellArg mod.version}
+    ( cd ${engine.src.name}; make version VERSION=${escapeShellArg engine.version} )
+  '';
+
+  makeFlags = "PREFIX=$(out)";
+
+  doCheck = true;
+  checkTarget = "test";
+
+  installPhase = ''
+    mkdir -p $out/lib/openra-${mod.name}
+    substitute ${./launch-game.sh} $out/lib/openra-${mod.name}/launch-game.sh \
+      --subst-var out \
+      --subst-var-by name ${escapeShellArg mod.name} \
+      --subst-var-by title ${escapeShellArg mod.title}
+    chmod +x $out/lib/openra-${mod.name}/launch-game.sh
+
+    # Setting TERM=xterm fixes an issue with terminfo in mono: System.Exception: Magic number is wrong: 542
+    # https://github.com/mono/mono/issues/6752#issuecomment-365212655
+    wrapProgram $out/lib/openra-${mod.name}/launch-game.sh \
+      --prefix PATH : "${path}" \
+      --prefix LD_LIBRARY_PATH : "${rpath}" \
+      --set TERM xterm
+
+    mkdir -p $out/bin
+    makeWrapper $out/lib/openra-${mod.name}/launch-game.sh $out/bin/openra-${mod.name} \
+      --run "cd $out/lib/openra-${mod.name}"
+
+    cp -r ${engine.src.name}/{${concatStringsSep "," [
```

Comment:
is it documented somewhere what files are needed?

Reply:
Nice, leveraging the Makefile of the engine, this code is no longer needed.